src/contexts/NotificationContext.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useReducer, useMemo } from 'react';
+import notificationReducer from '../reducers/notificationReducer';
+
+export const NotificationContext = createContext();
```

Comment:
Initialize createContext with the initial state:

```
const initialState = { notifications: [] };
export const NotificationContext = createContext(initialState);
```

See `UserListContext.jsx` as an example. Or simply set the initial state without declaring it:

```
export const NotificationContext = createContext({ notification: [] });
```

See `MessageContext.jsx` as an example.